File: Raven.Client.Lightweight/IDocumentSessionImpl.cs
//-----------------------------------------------------------------------
// <copyright file="IDocumentSessionImpl.cs" company="Hibernating Rhinos LTD">
//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
#if !SILVERLIGHT
using System;
using System.Collections.Generic;
using Raven.Client.Document;
using Raven.Client.Document.Batches;

namespace Raven.Client
{
	/// <summary>
	/// Interface for document session which holds the internal operations
	/// </summary>
	internal interface IDocumentSessionImpl : IDocumentSession, ILazySessionOperations, IEagerSessionOperations
	{
		DocumentConvention Conventions { get; }

		T[] LoadInternal<T>(string[] ids);
		T[] LoadInternal<T>(string[] ids, KeyValuePair<string, Type>[] includes);
		Lazy<T[]> LazyLoadInternal<T>(string[] ids, KeyValuePair<string, Type>[] includes, Action<T[]> onEval);
	}
}
#endif

File: Raven.Client.Java/lib/src/main/java/raven/client/IDocumentSessionImpl.java
package raven.client;

import raven.abstractions.basic.Lazy;
import raven.abstractions.basic.Tuple;
import raven.abstractions.closure.Action1;
import raven.client.document.DocumentConvention;
import raven.client.document.batches.IEagerSessionOperations;
import raven.client.document.batches.ILazySessionOperations;

/**
 * Interface for document session which holds the internal operations
 */
public interface IDocumentSessionImpl extends IDocumentSession, ILazySessionOperations, IEagerSessionOperations {

  public DocumentConvention getConventions();

  public <T> T[] loadInternal(Class<T> clazz, String[] ids);

  public <T> T[] loadInternal(Class<T> clazz, String[] ids, Tuple<String, Class<?>>[] includes);

  <T> Lazy<T[]> lazyLoadInternal(Class<T> clazz, String[] ids, Tuple<String, Class<?>>[] includes, Action1<T[]> onEval);
}
